### PR TITLE
[FIRRTL] Add utilities to NLATable analysis. NFC.

### PR DESCRIPTION
**Pull request by prithayan**
This commit adds utilities to NLATable analysis, 
 1. For getting the NLAs that an instance participates in.
 2. Remove an NLA from the Module.
 3. Remove a set of NLAs from the Module.

### CODE REVIEW

File: lib/Dialect/FIRRTL/NLATable.cpp [hunk 1]
```diff
@@ -119,11 +136,6 @@ FModuleLike NLATable::getModule(StringAttr name) {
   return dyn_cast_or_null<FModuleLike>(n);
 }
 
-// void NLATable::retargetInstance(InstanceOp inst, StringRef newModName) {
-//     inst.moduleNameAttr(FlatSymbolRefAttr::get(inst.getContext(),
-//     newModName));
-// }
-
```
Comment (dtzSiFive) [inline lib/Dialect/FIRRTL/NLATable.cpp:126]: Thanks for cleaning this up!

**Review by dtzSiFive — COMMENTED** [review]
Whoops, meant to interject last night.  Will make a PR in a few, want to fix a nit while at it.

**Comment by dtzSiFive** [comment]
(see #3181)

File: lib/Dialect/FIRRTL/NLATable.cpp [hunk 2]
```diff
@@ -88,6 +88,23 @@ void NLATable::renameModuleAndInnerRef(
   return;
 }
 
+void NLATable::removeNLAfromModule(NonLocalAnchor nla, StringAttr mod) {
+  auto nlaList = nodeMap[mod];
+  for (auto it = nlaList.begin(), e = nlaList.end(); it != e; ++it)
+    if ((*it) == nla) {
+      nlaList.erase(it);
```
[the remark this answers is not in the document]
Comment (prithayan) [reply lib/Dialect/FIRRTL/NLATable.cpp:95]: Thanks, your fix looks good. I agree with your other two suggestions here. Will make them on the next commit.

**Pull request merged**